### PR TITLE
MS Teams - Auto-resetting the token info when switching between authentication types

### PR DESCRIPTION
**Pull request by ShacharKidor**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-11414](https://jira-dc.paloaltonetworks.com/browse/CIAC-11414).
relates - [CIAC-11459 - Epic](https://jira-dc.paloaltonetworks.com/browse/CIAC-11459)

## Description
- Added an automatic run of the reset-auth function when a user switched the authentication type in the integration instance parameters (the new logic was added to the main function). 

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by ShacharKidor** [comment]
Notes regarding race conditions and edge cases were added to the JIRA ticket.

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml [hunk 1]
```diff
@@ -25,6 +25,9 @@ configuration:
   hidden: true
   required: false
 - display: Authentication Type
+  additionalinfo: |- 
+    When switching from the 'Client Credentials Flow' to the 'Authorization Code Flow' do as follow: Save the instance, run the 'microsoft-teams-generate-login-url' command and follow the instructions mentioned there. Once done, test the validity of your instance by running the 'microsoft-teams-auth-test' command.
+    When switching from the 'Authorization Code Flow' to the 'Client Credentials Flow', test the validity of the instance by clicking on the 'Test' button.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml:30]:
```suggestion
    When switching from the 'Authorization Code Flow' to the 'Client Credentials Flow', test the validity of the instance by clicking the 'Test' button.
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md [hunk 2]
```diff
@@ -1103,9 +1103,9 @@ There is no context output for this command.
 ***
 Run this command if for some reason you need to rerun the graph authentication process.
 Notes:
-- Use this command to switch between authentication flows and ensure the integration uses the appropriate token.
 - After making changes to permissions in the Azure Portal, reset the authentication to ensure that the token reflects the updated permissions.
-- When using the `Authorization Code Flow`, after executing the command, regenerate the **Authorization code** parameter, and then run the *!microsoft-teams-auth-test* command to verify the authentication.
+- This command is triggered automatically when an authentication flow type switch is detected. The auto resetting ensures the integration uses the appropriate token.
+- When switching the authentication type to the `Authorization Code Flow`, this command will be triggered automatically, after that you will need to regenerate the **Authorization code** parameter by running the ***microsoft-teams-generate-login-url*** command, and to verify the authentication by running the ***!microsoft-teams-auth-test*** command.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md:1108]:
```suggestion
- When switching the authentication type to the `Authorization Code Flow`, this command will be triggered automatically. Then you will need to regenerate the **Authorization code** parameter by running the ***microsoft-teams-generate-login-url*** command, and to verify the authentication by running the ***!microsoft-teams-auth-test*** command.
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md [hunk 3]
```diff
@@ -1225,7 +1225,8 @@ Note: To enrich an incident created via the Demisto BOT (`new incident` command)
...
+Compare the permissions list obtained for the token with the permissions required for the command you wish to execute (can be found in the command documentation). If there are missing API permissions, add them to your application, and then run the `microsoft-teams-auth-reset` command (as described here - [microsoft-teams-auth-reset documentation](https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#microsoft-teams-auth-reset)). 
+If your authentication type is the `Authorization Code Flow` after running the `microsoft-teams-auth-reset` command you will need to regenerate the **Authorization code** parameter by running the ***microsoft-teams-generate-login-url*** command, and to verify the authentication by running the ***!microsoft-teams-auth-test*** command.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md:1229]:
```suggestion
If your authentication type is the `Authorization Code Flow`, after running the `microsoft-teams-auth-reset` command you will need to regenerate the **Authorization code** parameter by running the ***microsoft-teams-generate-login-url*** command, and to verify the authentication by running the ***!microsoft-teams-auth-test*** command.
```

File: Packs/MicrosoftTeams/ReleaseNotes/1_5_9.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Microsoft Teams
+
+- You can now switch the *Authentication Type* of your instance without manually running the ***microsoft-teams-auth-reset*** command. The reset authentication logic will be triggered automatically when an authentication switch is occurred.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftTeams/ReleaseNotes/1_5_9.md:6]:
```suggestion
You can now switch the *Authentication Type* of your instance without manually running the ***microsoft-teams-auth-reset*** command. The reset authentication logic is triggered automatically when an authentication switch occurs.
```

**Comment by ShirleyDenkberg** [comment]
@kobymeir @yuvalbenshalom @merit-maita @michal-dagan Doc review completed.

**Review by yuvalbenshalom — COMMENTED** [review]
looks good. please have this docs reviewed.

**Comment by ShacharKidor** [comment]
@yuvalbenshalom - docs are already approved by @ShirleyDenkberg.

**Review by yuvalbenshalom — APPROVED** [review]

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**